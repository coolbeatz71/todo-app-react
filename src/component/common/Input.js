import React from 'react';
import PropTypes from 'prop-types';

/**
 * Input component
 * @param {object} props
 * @returns {void}
 */
const Input = ({ placeholder, name, onChange, type, myClass, value }) => (
  <input
    type={type}
    name={name}
    value={value}
    placeholder={placeholder}
    onChange={onChange}
    className={`text-input ${myClass}`}
  />
);

Input.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  myClass: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

Input.defaultProps = {
  name: 'title',
  type: 'text',
  placeholder: 'Enter task to the TO-DO list',
  myClass: '',
};

export default Input;
