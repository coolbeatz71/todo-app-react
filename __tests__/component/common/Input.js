import React from 'react';
import { shallow } from 'enzyme';
import Input from '../../../src/component/common/Input';

const props = {
  value: 'just for testing',
  onChange: jest.fn(),
};
describe('Input.js', () => {
  test('should render the input component', () => {
    const wrapper = shallow(<Input {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
