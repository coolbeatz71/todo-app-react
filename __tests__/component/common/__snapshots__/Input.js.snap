// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.jsx should render the input component 1`] = `ShallowWrapper {}`;
